src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts

```diff
@@ -115,17 +127,119 @@ describe('hasShortWaitTimeAndNotGuaranteedCorrespondence', () => {
+  });
+
+  it('should be false despite short wait time between second and third leg, beacuse interchange is guarenteed first leg and mode equals foot in second leg.', () => {
+    const legs = [
+      {
+        expectedStartTime: '2024-10-31T14:45:00Z',
+        expectedEndTime: '2024-10-31T15:00:00Z',
+        interchangeTo: {guaranteed: true},
+      },
       {
         mode: 'foot',
-        expectedStartTime: '2024-10-31T15:25:00Z',
-        expectedEndTime: '2024-10-31T15:30:00Z',
+        expectedStartTime: '2024-10-31T15:00:00Z',
+        expectedEndTime: '2024-10-31T15:05:00Z',
       },
       {
-        expectedStartTime: '2024-10-31T15:31:00Z',
-        expectedEndTime: '2024-10-31T15:40:00Z',
+        expectedStartTime: '2024-10-31T15:06:00Z',
+        expectedEndTime: '2024-10-31T15:20:00Z',
       },
     ] as Leg[];
 
     expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(false);
   });
+
+  it('should be true because because of short wait time between second and third leg, and the mode in the second leg is not foot.', () => {
+    const legs = [
+      {
+        expectedStartTime: '2024-10-31T14:45:00Z',
+        expectedEndTime: '2024-10-31T15:00:00Z',
+        interchangeTo: {guaranteed: true},
+      },
+      {
+        mode: 'bus',
+        expectedStartTime: '2024-10-31T15:00:00Z',
+        expectedEndTime: '2024-10-31T15:05:00Z',
+      },
+      {
+        expectedStartTime: '2024-10-31T15:06:00Z',
+        expectedEndTime: '2024-10-31T15:20:00Z',
+      },
+    ] as Leg[];
+
+    expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(true);
+  });
+
+  it('should be true because of short wait time and no guarenteed correspondance.', () => {
+    const legs = [
+      {
+        mode: 'foot',
+        expectedStartTime: '2024-10-31T14:45:00Z',
+        expectedEndTime: '2024-10-31T15:00:00Z',
+      },
+      {
+        expectedStartTime: '2024-10-31T15:00:00Z',
+        expectedEndTime: '2024-10-31T15:05:00Z',
+      },
+    ] as Leg[];
+
+    expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(true);
+  });
+
+  it('should be true because of short wait time between third and forth leg.', () => {
+    const legs = [
+      {
+        expectedStartTime: '2024-10-31T14:45:00Z',
+        expectedEndTime: '2024-10-31T15:00:00Z',
+        interchangeTo: {guaranteed: true},
+      },
+      {
+        mode: 'foot',
+        expectedStartTime: '2024-10-31T15:00:00Z',
+        expectedEndTime: '2024-10-31T15:05:00Z',
+      },
+
+      {
+        expectedStartTime: '2024-10-31T15:06:00Z',
+        expectedEndTime: '2024-10-31T15:20:00Z',
+      },
+      {
+        expectedStartTime: '2024-10-31T15:21:00Z',
+        expectedEndTime: '2024-10-31T15:30:00Z',
+      },
+    ] as unknown as Leg[];
+
+    expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(true);
+  });
+
+  it('should be true because of short wait time between third and forth leg.', () => {
```

Comment:
Shouldn't have two tests with the same name

Reply:
Okey, i´ll change the name👍